### ISSUE TITLE
thread '<unknown>' has overflowed its stack

### ISSUE THREAD

**Issue by Lapz** [rust-analyzer/rust-analyzer#587]
When trying to hover on a def after #586 I get the following error 
```
Request textDocument/hover failed.
  Message: content modified
  Code: -32801 

thread '<unknown>' has overflowed its stack
fatal runtime error: stack overflow
```

**Comment by Lapz** [comment]
It happended when I was hovering over an enum varinat name i.e. 
```rust 
Result::Ok
```

**Comment by flodiebold** [comment]
Hm, I can't reproduce it so far. Do you have a specific file or project where this happens? Does it happen all the time or just sometimes?

**Comment by flodiebold** [comment]
@Lapz does #686 fix the problem?

**Comment by flodiebold** [comment]
If it wasn't fixed by #686, I'm pretty sure it was fixed by #773, so closing this...
